pkg/exprparser/interpreter.go

```diff
@@ -20,7 +20,7 @@ type EvaluationEnvironment struct {
 	Secrets  map[string]string
 	Strategy map[string]interface{}
 	Matrix   map[string]interface{}
-	Needs    map[string]map[string]map[string]string
+	Needs    map[string]map[string]interface{}
```

Comment:
Code looks fine, but I would prefer if this uses a more concrete type.
The actual type would be `map[string]Needs`
with `Needs` like this:

```go
type Needs struct {
  outputs map[string]string
  result  string
}
```

Isn't it?

Reply:
> I would prefer if this uses a more concrete type.

Me too!

> The actual type would be `map[string]Needs` with `Needs` like this:
> 
> ```go
> type Needs struct {
>   outputs map[string]string
>   result  string
> }
> ```

Thanks for this. I'm not familiar with Go, so wasn't sure what to do about this. I've pushed the change.